frontend/src/metabase/metadata/pages/DataModel/components/PreviewSection/PreviewSection.tsx

```diff
@@ -59,7 +68,19 @@ export const PreviewSection = ({
         />
       )}
 
-      {previewType === "filtering" && <Box>FILTERING</Box>}
+      {previewType === "filtering" && (
+        <FilteringPreview
+          databaseId={databaseId}
+          fieldId={fieldId}
+          /**
+           * Make sure internal component state is reset when changing any field settings.
+           * This is because use***Filter hooks cache some parts of state internally on mount
+           * and do not account for all prop changes during their lifecycle.
+           */
+          key={JSON.stringify(field)}
```

Comment:
Should we use a stable stringifier or memoize the value to avoid unnecessary re-renders?

We already depend on `fast-json-stable-stringify` so maybe you could use that?